test/unit/kubernetes-deploy/kubernetes_resource_test.rb

```diff
@@ -321,7 +321,7 @@ def test_build_handles_hardcoded_and_core_and_dynamic_objects
     redis_cr = KubernetesDeploy::KubernetesResource.build(namespace: "test", context: "test",
       logger: @logger, statsd_tags: [], crd: redis_crd,
       definition: { "kind" => "Redis", "metadata" => { "name" => "test" } })
-    assert_equal(redis_cr.class, KubernetesDeploy::Redis)
+    assert_equal(redis_cr.class, KubernetesDeploy::CustomResource)
```

Comment:
This assertion shouldn't be changed to the generic class--this part of the test is proving that if a hardcoded class exists, it takes precedence. Therefore, Redis can't be used for it anymore. Simply change the example to use Cloudsql or Elasticsearch, which are our last to hardcoded Shopify CRs.